hs_discover/src/components/Resources.vue

```diff
@@ -399,6 +404,27 @@ export default {
       }
       return [];
     },
+    filterBuilder() {
```

Comment:
hmm, so it looks like filterbuilder is called every text search.  The same call is made again but with the filterbuilder flag.  So, I'm not understanding why it's going through the effort of building the full response instead of just returning when the filterbuilder is ready.

Reply:
The expected behavior, based on how Prod HydroShare Discover works, is that each time you freetext search (and other operations), the SearchQueryset returns a subset of matching resources, and the facets update to reflect that resource count / properties, such as Owner, for example.

Reply:
I was not clear above.  I understand the functionality.  

You make two calls to this endpoint, one with the filterbuilder parameter and one without.  Filterbuilder takes longer so it makes sense to make it asynchronous.  

I'm asking why the call with the filterbuilder parameter also includes the data that is already built and provided in the first call (without the filterbuilder parameter). This function discards all fields outside of the filterdata key.

Reply:
You're referring to two calls to the endpoint. That was the old method. As of this New Discover #4007 PR, there's only one call to the /discoverapi endpoint on page load. Each action, freetext search, temporal filter or map mode change also calls /discoverapi with a filter rebuild (redo the checkbox counts). Hopefully that answers the question if not I'm happy to continue discussing here.

Reply:
ok, I see what's going on now.  filterbuilder is included on initial searches, but left out on sorts.  Thanks.

Reply:
Is this function used anywhere?  I don't think it is... the only reference to it that I see is commented out.

Reply:
I think it can be removed.

Reply:
It should be removed, not sure why it was added if it was never used in this PR.  It's what threw me off, made me think there were two calls to the backend again.